From c3a8d253e7e4086d0ac9ca8415686538d5f1f065 Mon Sep 17 00:00:00 2001
From: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Date: Tue, 23 Apr 2024 15:19:17 +0530
Subject: [PATCH] drivers: qcom_scm: Update argument list to pass TZ

Add fix to update correct argument list to pass TZ

Signed-off-by: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Change-Id: I033ca688633f70c5ce2de73070f67e06a7ef9f3c
---
 drivers/firmware/qcom_scm.c | 2 +-
 drivers/soc/qcom/ice.c      | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/firmware/qcom_scm.c b/drivers/firmware/qcom_scm.c
index 26cd25f10e33..450c4d332f1a 100644
--- a/drivers/firmware/qcom_scm.c
+++ b/drivers/firmware/qcom_scm.c
@@ -321,7 +321,7 @@ int qcom_config_sec_ice(void *buf, int size)
 	struct qcom_scm_desc desc = {
 		.svc = QCOM_SVC_ICE,
 		.cmd = QCOM_SCM_ICE_CMD,
-		.arginfo = QCOM_SCM_ARGS(2),
+		.arginfo = QCOM_SCM_ARGS(2, QCOM_SCM_RO),
 		.owner = ARM_SMCCC_OWNER_SIP,
 	};
 
diff --git a/drivers/soc/qcom/ice.c b/drivers/soc/qcom/ice.c
index e61dd57ada53..d265f77a37bf 100644
--- a/drivers/soc/qcom/ice.c
+++ b/drivers/soc/qcom/ice.c
@@ -172,7 +172,7 @@ static int sdhci_msm_ice_set_hwkey_config(struct qcom_ice *ice,
 	struct ice_config_sec *ice_settings = NULL;
 	int ret;
 
-	ice_settings = kmalloc(sizeof(struct ice_config_sec), GFP_KERNEL);
+	ice_settings = kzalloc(sizeof(struct ice_config_sec), GFP_KERNEL);
 	if (!ice)
 		return -ENOMEM;
 
-- 
2.34.1

